packages/camera/camera_android_camerax/pigeons/camerax_library.dart

```diff
@@ -279,298 +418,858 @@ abstract class DeviceOrientationManagerHostApi {
+  void close();
 
-  int getTargetVideoEncodingBitRate(int identifier);
+  /// Pauses the current recording if active.
+  void pause();
 
-  int prepareRecording(int identifier, String path);
-}
+  /// Resumes the current recording if paused.
+  void resume();
 
-@FlutterApi()
-abstract class RecorderFlutterApi {
-  void create(int identifier, int? aspectRatio, int? bitRate);
+  /// Stops the recording, as if calling `close`.
+  ///
+  /// This method is equivalent to calling `close`.
+  void stop();
 }
 
-@HostApi(dartHostTestHandler: 'TestPendingRecordingHostApi')
-abstract class PendingRecordingHostApi {
-  int start(int identifier);
+/// FlashModes for image capture.
+///
+/// See https://developer.android.com/reference/kotlin/androidx/camera/core/ImageCapture#FLASH_MODE_AUTO().
+enum CameraXFlashMode {
+  /// Auto flash.
+  ///
+  /// The flash will be used according to the camera system's determination when
+  /// taking a picture.
+  auto,
+
+  /// No flash.
+  ///
+  /// The flash will never be used when taking a picture.
+  off,
+
+  /// Always flash.
+  ///
+  /// The flash will always be used when taking a picture.
+  on,
 }
 
-@FlutterApi()
-abstract class PendingRecordingFlutterApi {
-  void create(int identifier);
+/// A use case for taking a picture.
+///
+/// See https://developer.android.com/reference/kotlin/androidx/camera/core/ImageCapture.
+@ProxyApi(
+  kotlinOptions: KotlinProxyApiOptions(
+    fullClassName: 'androidx.camera.core.ImageCapture',
+  ),
+)
+abstract class ImageCapture extends UseCase {
+  ImageCapture(int? targetRotation, CameraXFlashMode? flashMode);
+
+  late final ResolutionSelector? resolutionSelector;
 
-  void onVideoRecordingEvent(VideoRecordEventData event);
+  /// Set the flash mode.
+  void setFlashMode(CameraXFlashMode flashMode);
+
+  /// Captures a new still image for in memory access.
+  @async
+  String takePicture();
+
+  /// Sets the desired rotation of the output image.
+  void setTargetRotation(int rotation);
 }
 
-@HostApi(dartHostTestHandler: 'TestRecordingHostApi')
-abstract class RecordingHostApi {
-  void close(int identifier);
+/// Fallback rule for choosing an alternate size when the specified bound size
+/// is unavailable.
+///
+/// See https://developer.android.com/reference/kotlin/androidx/camera/core/resolutionselector/ResolutionStrategy.
+enum ResolutionStrategyFallbackRule {
+  /// When the specified bound size is unavailable, CameraX falls back to the
+  /// closest higher resolution size.
+  closestHigher,
+
+  /// When the specified bound size is unavailable, CameraX falls back to select
+  /// the closest higher resolution size.
+  closestHigherThenLower,
+
+  /// When the specified bound size is unavailable, CameraX falls back to the
+  /// closest lower resolution size.
+  closestLower,
 
-  void pause(int identifier);
+  /// When the specified bound size is unavailable, CameraX falls back to select
+  /// the closest lower resolution size.
+  closestLowerThenHigher,
```

Comment:
Is this right? It's the same as above.

Reply:
It looks like I only copied the first sentence from the public docs. I added the second since it seems important:

https://developer.android.com/reference/kotlin/androidx/camera/core/resolutionselector/ResolutionStrategy#FALLBACK_RULE_CLOSEST_LOWER_THEN_HIGHER()